Assets/Game/Scripts/PlugableAI/Addon/Sight.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using UnityEngine;
+
+[CreateAssetMenu(menuName = "PluggableAI/Addon/Sight"), Serializable]
+public class Sight: ScriptableObject
+{
+  [SerializeField] private float radius;
+  [TagSelector, SerializeField] private string targetTag;
+
+  public bool SearchTarget(StateController controller)
```

Comment:
How does the search logic work? Does it search for player's collider?

Reply:
The logic is not decided, right now it just uses Physics.OverlapSphere to search for collider that has the same tag as `targetTag` within `radius` and set the collider's game object to controller's `target`.